java/statestore/src/main/java/sleeper/statestore/s3/S3FileReferenceStore.java

```diff
@@ -228,16 +227,15 @@ public void atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFiles(
                 if (filesToBeMarkedReadyForGCSet.contains(existingFile.getFilename())) {
                     file = file.removeReferenceForPartition(partitionId, updateTime);
                 }
-                AllReferencesToAFile newFile = newFilesByName.get(existingFile.getFilename());
-                if (newFile != null) {
-                    file = file.addReferences(newFile.getInternalReferences(), updateTime);
+                if (existingFile.getFilename().equals(newFile.getFilename())) {
+                    file = file.addReferences(List.of(newFile), updateTime);
```

Comment:
If we disallow using the same filename for the old files and the new one, this logic won't be needed.